pkg/sdk/data_types_deprecated.go

```diff
@@ -0,0 +1,51 @@
+package sdk
+
+import (
+	"strings"
+
+	"github.com/Snowflake-Labs/terraform-provider-snowflake/pkg/sdk/datatypes"
+)
+
+// DataType is based on https://docs.snowflake.com/en/sql-reference/intro-summary-data-types.
+type DataType string
+
+var allowedVectorInnerTypes = []DataType{
+	DataTypeInt,
+	DataTypeFloat,
+}
+
+const (
+	DataTypeNumber       DataType = "NUMBER"
+	DataTypeInt          DataType = "INT"
+	DataTypeFloat        DataType = "FLOAT"
+	DataTypeVARCHAR      DataType = "VARCHAR"
+	DataTypeString       DataType = "STRING"
+	DataTypeBinary       DataType = "BINARY"
+	DataTypeBoolean      DataType = "BOOLEAN"
+	DataTypeDate         DataType = "DATE"
+	DataTypeTime         DataType = "TIME"
+	DataTypeTimestampLTZ DataType = "TIMESTAMP_LTZ"
+	DataTypeTimestampNTZ DataType = "TIMESTAMP_NTZ"
+	DataTypeTimestampTZ  DataType = "TIMESTAMP_TZ"
+	DataTypeVariant      DataType = "VARIANT"
+	DataTypeObject       DataType = "OBJECT"
+	DataTypeArray        DataType = "ARRAY"
+	DataTypeGeography    DataType = "GEOGRAPHY"
+	DataTypeGeometry     DataType = "GEOMETRY"
+)
+
+// IsStringType is a legacy method. datatypes.IsTextDataType should be used instead.
+// TODO [SNOW-1348114]: remove with tables rework
```

Comment:
No, because then linter fails (so make pre-push fails), I already wrote about it here: https://github.com/Snowflake-Labs/terraform-provider-snowflake/pull/3207 (search for deprecated in PR description)